app/src/main/java/app/revanced/manager/ui/component/patches/OptionFields.kt

```diff
@@ -141,26 +156,37 @@ private inline fun <T : Any> WithOptionEditor(
     option: Option<T>,
     value: T?,
     noinline setValue: (T?) -> Unit,
+    selectionWarningEnabled: Boolean,
     crossinline onDismissDialog: @DisallowComposableCalls () -> Unit = {},
     block: OptionEditorScope<T>.() -> Unit
 ) {
     var showDialog by rememberSaveable { mutableStateOf(false) }
+    var showSelectionWarning by rememberSaveable { mutableStateOf(false) }
     val scope = remember(editor, option, value, setValue) {
         OptionEditorScope(
             editor,
             option,
             openDialog = { showDialog = true },
+            clickAction = { clickAction ->
+                if (!option.required && selectionWarningEnabled)
+                    showSelectionWarning = true
+                else
+                    clickAction()
+            },
```

Comment:
`OptionEditorScope` has all the necessary information in order to make that decision. I think it would be more readable to move this logic inside `OptionEditorScope` and supply it with a callback to open the dialog instead of having a callback that accepts another callback. This is already done with `openDialog`.